### DISCUSSION TITLE
How to activate Recovery Code v18?

### DISCUSSION THREAD

**Discussion by masterbater** [keycloak/keycloak#11887] [Q&A] [answered]
https://www.keycloak.org/docs/latest/server_admin/#_recovery-codes
Reading from this docs  is lacking. I dont even see it when creating new flows. Is this preview feature only activated during build?

**Comment by darius-m** [comment, marked as answer]
In the legacy configuration you can see how to enable features in the [Profiles](https://www.keycloak.org/docs/latest/server_installation/index.html#profiles) section. For the Quarkus distribution, I think you will need to use the [Features](https://www.keycloak.org/server/all-config#_feature) option.

**Reply by masterbater** [reply]
Thanks I manage to activate it, I added it as conditional 2fa, it shows up in profile 
![image](https://user-images.githubusercontent.com/26339988/167782822-89383845-df7b-48ed-8fd3-a1c7881c7f55.png)

My only issue is that during login, only otp is being  shown no alternative input for recovery code
![image](https://user-images.githubusercontent.com/26339988/167782919-f0f72039-b0ea-442f-8883-89a003c58be7.png)

**Reply by masterbater** [reply]
browser flow 
![image](https://user-images.githubusercontent.com/26339988/167783207-9589d71a-bc91-4209-8dce-3354434e5425.png)

**Reply by darius-m** [reply]
I have not yet used the recovery codes, but I assume they are similar to how they are implemented on other sites. In this case, you can either use the OTP generated by the application, or one of the recovery codes. Once a recovery code has been used, it cannot be used again, but they will not expire otherwise, as opposed to the regular OTP codes.
